From 73e6b8e18418bdab52366cdf8444386eb3c2676d Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Tue, 6 Sep 2022 16:39:04 +0800
Subject: [PATCH] sw64: sort Kconfig select
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 73e6b8e18418bdab52366cdf8444386eb3c2676d
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5XTGY

--------------------------------

Signed-off-by: Mao Minkai <maominkai@wxiat.com>
Reviewed-by: He Sheng <hesheng@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/Kconfig | 136 ++++++++++++++++++++++-----------------------
 1 file changed, 68 insertions(+), 68 deletions(-)

diff --git a/arch/sw_64/Kconfig b/arch/sw_64/Kconfig
index 36cddefb1ad4..7174b6218539 100644
--- a/arch/sw_64/Kconfig
+++ b/arch/sw_64/Kconfig
@@ -2,40 +2,13 @@
 config SW64
 	bool
 	default y
-	select AUDIT_ARCH
-	select HAVE_IDE
-	select HAVE_OPROFILE
-	select HAVE_PCSPKR_PLATFORM
-	select HAVE_PERF_EVENTS
-	select HAVE_FAST_GUP
-	select GENERIC_CLOCKEVENTS
-	select GENERIC_IRQ_PROBE
-	select GENERIC_IRQ_LEGACY
-	select GENERIC_IRQ_SHOW
-	select ARCH_WANT_IPC_PARSE_VERSION
-	select ARCH_HAVE_NMI_SAFE_CMPXCHG
-	select ARCH_NO_PREEMPT
-	select ARCH_USE_CMPXCHG_LOCKREF
-	select GENERIC_SMP_IDLE_THREAD
-	select HAVE_MOD_ARCH_SPECIFIC
-	select MODULES_USE_ELF_RELA
-	select ARCH_SUPPORTS_NUMA_BALANCING
-	select HAVE_ARCH_TRANSPARENT_HUGEPAGE
-	select HAVE_ARCH_AUDITSYSCALL
-	select HAVE_ARCH_SECCOMP_FILTER
-	select OLD_SIGSUSPEND
-	select GENERIC_STRNCPY_FROM_USER
-	select GENERIC_STRNLEN_USER
-	select HAVE_ARCH_KGDB
+	select ACPI
+	select ACPI_REDUCED_HARDWARE_ONLY
+	select ARCH_HAS_ELF_RANDOMIZE
 	select ARCH_HAS_PHYS_TO_DMA
-	select SWIOTLB
-	select HAVE_MEMBLOCK
-	select HAVE_MEMBLOCK_NODE_MAP
-	select NO_BOOTMEM
-	select ARCH_USE_QUEUED_RWLOCKS
-	select ARCH_USE_QUEUED_SPINLOCKS
-	select COMMON_CLK
-	select HANDLE_DOMAIN_IRQ
+	select ARCH_HAS_PTE_SPECIAL
+	select ARCH_HAS_SG_CHAIN
+	select ARCH_HAVE_NMI_SAFE_CMPXCHG
 	select ARCH_INLINE_READ_LOCK
 	select ARCH_INLINE_READ_LOCK_BH
 	select ARCH_INLINE_READ_LOCK_IRQ
@@ -44,60 +17,87 @@ config SW64
 	select ARCH_INLINE_READ_UNLOCK_BH
 	select ARCH_INLINE_READ_UNLOCK_IRQ
 	select ARCH_INLINE_READ_UNLOCK_IRQRESTORE
-	select ARCH_INLINE_WRITE_LOCK
-	select ARCH_INLINE_WRITE_LOCK_BH
-	select ARCH_INLINE_WRITE_LOCK_IRQ
-	select ARCH_INLINE_WRITE_LOCK_IRQSAVE
-	select ARCH_INLINE_WRITE_UNLOCK
-	select ARCH_INLINE_WRITE_UNLOCK_BH
-	select ARCH_INLINE_WRITE_UNLOCK_IRQ
-	select ARCH_INLINE_WRITE_UNLOCK_IRQRESTORE
-	select ARCH_INLINE_SPIN_TRYLOCK
-	select ARCH_INLINE_SPIN_TRYLOCK_BH
 	select ARCH_INLINE_SPIN_LOCK
 	select ARCH_INLINE_SPIN_LOCK_BH
 	select ARCH_INLINE_SPIN_LOCK_IRQ
 	select ARCH_INLINE_SPIN_LOCK_IRQSAVE
+	select ARCH_INLINE_SPIN_TRYLOCK
+	select ARCH_INLINE_SPIN_TRYLOCK_BH
 	select ARCH_INLINE_SPIN_UNLOCK
 	select ARCH_INLINE_SPIN_UNLOCK_BH
 	select ARCH_INLINE_SPIN_UNLOCK_IRQ
 	select ARCH_INLINE_SPIN_UNLOCK_IRQRESTORE
+	select ARCH_INLINE_WRITE_LOCK
+	select ARCH_INLINE_WRITE_LOCK_BH
+	select ARCH_INLINE_WRITE_LOCK_IRQ
+	select ARCH_INLINE_WRITE_LOCK_IRQSAVE
+	select ARCH_INLINE_WRITE_UNLOCK
+	select ARCH_INLINE_WRITE_UNLOCK_BH
+	select ARCH_INLINE_WRITE_UNLOCK_IRQ
+	select ARCH_INLINE_WRITE_UNLOCK_IRQRESTORE
+	select ARCH_NO_PREEMPT
+	select ARCH_SUPPORTS_ACPI
 	select ARCH_SUPPORTS_ATOMIC_RMW
-	select ARCH_HAS_SG_CHAIN
-	select IRQ_FORCED_THREADING
+	select ARCH_SUPPORTS_NUMA_BALANCING
+	select ARCH_SUPPORTS_UPROBES
+	select ARCH_USE_CMPXCHG_LOCKREF
+	select ARCH_USE_QUEUED_RWLOCKS
+	select ARCH_USE_QUEUED_SPINLOCKS
+	select ARCH_WANT_FRAME_POINTERS
+	select ARCH_WANT_IPC_PARSE_VERSION
+	select AUDIT_ARCH
+	select COMMON_CLK
+	select DMA_OPS if PCI
+	select GENERIC_CLOCKEVENTS
+	select GENERIC_IRQ_LEGACY
 	select GENERIC_IRQ_MIGRATION if SMP
+	select GENERIC_IRQ_PROBE
+	select GENERIC_IRQ_SHOW
+	select GENERIC_PCI_IOMAP if PCI
+	select GENERIC_SMP_IDLE_THREAD
+	select GENERIC_STRNCPY_FROM_USER
+	select GENERIC_STRNLEN_USER
+	select GENERIC_TIME_VSYSCALL
+	select HANDLE_DOMAIN_IRQ
+	select HARDIRQS_SW_RESEND
+	select HAVE_ARCH_AUDITSYSCALL
+	select HAVE_ARCH_JUMP_LABEL
+	select HAVE_ARCH_KGDB
+	select HAVE_ARCH_SECCOMP_FILTER
 	select HAVE_ARCH_TRACEHOOK
-	select HAVE_FUNCTION_TRACER
+	select HAVE_ARCH_TRANSPARENT_HUGEPAGE
+	select HAVE_ASM_MODVERSIONS
+	select HAVE_C_RECORDMCOUNT
 	select HAVE_DYNAMIC_FTRACE
+	select HAVE_EBPF_JIT
+	select HAVE_FAST_GUP
 	select HAVE_FTRACE_MCOUNT_RECORD
-	select HAVE_C_RECORDMCOUNT
 	select HAVE_FUNCTION_GRAPH_TRACER
+	select HAVE_FUNCTION_TRACER
+	select HAVE_IDE
 	select HAVE_KPROBES
 	select HAVE_KRETPROBES
-	select HAVE_SYSCALL_TRACEPOINTS
-	select ARCH_SUPPORTS_UPROBES
-	select OF_EARLY_FLATTREE if OF
-	select HAVE_EBPF_JIT
-	select SPARSEMEM_EXTREME if SPARSEMEM
-	select HAVE_ARCH_JUMP_LABEL
-	select ARCH_WANT_FRAME_POINTERS
-	select HAVE_ASM_MODVERSIONS
-	select ARCH_HAS_ELF_RANDOMIZE
-	select HAVE_PERF_USER_STACK_DUMP
-	select HAVE_PERF_REGS
-	select ARCH_SUPPORTS_ACPI
-	select ACPI
-	select ACPI_REDUCED_HARDWARE_ONLY
-	select GENERIC_TIME_VSYSCALL
-	select SET_FS
+	select HAVE_MEMBLOCK
+	select HAVE_MEMBLOCK_NODE_MAP
+	select HAVE_MOD_ARCH_SPECIFIC
+	select HAVE_OPROFILE
 	select HAVE_PCI
-	select GENERIC_PCI_IOMAP if PCI
-	select PCI_MSI_ARCH_FALLBACKS
-	select DMA_OPS if PCI
+	select HAVE_PCSPKR_PLATFORM
+	select HAVE_PERF_EVENTS
+	select HAVE_PERF_REGS
+	select HAVE_PERF_USER_STACK_DUMP
 	select HAVE_REGS_AND_STACK_ACCESS_API
-	select ARCH_HAS_PTE_SPECIAL
-	select HARDIRQS_SW_RESEND
+	select HAVE_SYSCALL_TRACEPOINTS
+	select IRQ_FORCED_THREADING
 	select MEMORY_HOTPLUG_SPARSE if MEMORY_HOTPLUG
+	select MODULES_USE_ELF_RELA
+	select NO_BOOTMEM
+	select OF_EARLY_FLATTREE if OF
+	select OLD_SIGSUSPEND
+	select PCI_MSI_ARCH_FALLBACKS
+	select SET_FS
+	select SPARSEMEM_EXTREME if SPARSEMEM
+	select SWIOTLB
 
 config LOCKDEP_SUPPORT
 	def_bool y
-- 
2.33.0

